src/snowflake/cli/plugins/git/commands.py

```diff
@@ -0,0 +1,236 @@
+import logging
+from pathlib import Path
+
+import typer
+from click import ClickException
+from snowflake.cli.api.commands.flags import (
+    identifier_argument,
+    like_option,
+    pattern_option,
+)
+from snowflake.cli.api.commands.snow_typer import SnowTyper
+from snowflake.cli.api.console.console import cli_console
+from snowflake.cli.api.constants import ObjectType
+from snowflake.cli.api.output.types import CommandResult, QueryResult
+from snowflake.cli.api.utils.path_utils import is_stage_path
+from snowflake.cli.plugins.git.manager import GitManager
+from snowflake.cli.plugins.object.manager import ObjectManager
+
+app = SnowTyper(
+    name="git",
+    help="Manages git repositories in Snowflake.",
+)
+log = logging.getLogger(__name__)
+
+
+def _repo_path_argument_callback(path):
+    if not is_stage_path(path):
+        raise ClickException(
+            "REPOSITORY_PATH should be a path to git repository stage with scope provided."
+            " For example: @my_repo/branches/main/"
+        )
+    return path
+
+
+RepoNameArgument = identifier_argument(sf_object="git repository", example="my_repo")
+RepoPathArgument = typer.Argument(
+    metavar="REPOSITORY_PATH",
+    help=(
+        "Path to git repository stage with scope provided."
+        " Path to the repository root must end with '/'."
+        " For example: @my_repo/branches/main/"
+    ),
+    callback=_repo_path_argument_callback,
+)
+PatternOption = pattern_option(
+    help_example='`list-files --pattern ".*\.txt"` lists all files with .txt extension',
+)
+
+
+def _assure_repository_does_not_exist(repository_name: str) -> None:
+    if ObjectManager().object_exists(
+        object_type=ObjectType.GIT_REPOSITORY.value.cli_name, name=repository_name
+    ):
+        raise ClickException(f"Repository '{repository_name}' already exists")
+
+
+def _validate_origin_url(url: str) -> None:
+    if not url.startswith("https://"):
+        raise ClickException("Url address should start with 'https'")
+
+
+@app.command("setup", requires_connection=True)
+def setup(
+    repository_name: str = RepoNameArgument,
+    **options,
+) -> CommandResult:
+    """
+    Sets up a git repository object.
+
+    You will be prompted for:
+
+    * url - address of repository to be used for git clone operation
+
+    * secret - Snowflake secret containing authentication credentials. Not needed if origin repository does not require
+    authentication for RO operations (clone, fetch)
+
+    * API integration - object allowing Snowflake to interact with git repository.
+    """
+    _assure_repository_does_not_exist(repository_name)
+    manager = GitManager()
+    om = ObjectManager()
```

Comment:
We create and use this manger later on, why not reuse it also in `_assure_repository_does_not_exist`?

Reply:
done